instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriver.java

```diff
@@ -0,0 +1,223 @@
+  }
+
+  /**
+   * Register the driver against {@link DriverManager}. This is done automatically when the class is
+   * loaded. Dropping the driver from DriverManager's list is possible using {@link #deregister()}
+   * method.
+   *
+   * @throws IllegalStateException if the driver is already registered
+   * @throws SQLException if registering the driver fails
+   */
+  public static void register() throws SQLException {
+    if (isRegistered()) {
+      throw new IllegalStateException(
+          "Driver is already registered. It can only be registered once.");
+    }
+    DriverManager.registerDriver(INSTANCE);
+    OpenTelemetryDriver.registered = true;
+  }
+
+  /**
+   * According to JDBC specification, this driver is registered against {@link DriverManager} when
+   * the class is loaded. To avoid leaks, this method allow unregistering the driver so that the
+   * class can be gc'ed if necessary.
+   *
+   * @throws IllegalStateException if the driver is not registered
+   * @throws SQLException if deregistering the driver fails
+   */
+  public static void deregister() throws SQLException {
+    if (!registered) {
+      throw new IllegalStateException(
+          "Driver is not registered (or it has not been registered using Driver.register() method)");
+    }
+    DriverManager.deregisterDriver(INSTANCE);
+    registered = false;
+  }
+
+  /** Returns {@code true} if the driver is registered against {@link DriverManager} */
+  public static boolean isRegistered() {
+    return registered;
+  }
+
+  private static Driver findDriver(String realUrl) throws SQLException {
+    if (realUrl == null || realUrl.trim().length() == 0) {
+      throw new IllegalArgumentException("url is required");
+    }
+
+    for (Driver candidate : Collections.list(DriverManager.getDrivers())) {
+      try {
+        if (!(candidate instanceof OpenTelemetryDriver) && candidate.acceptsURL(realUrl)) {
+          return candidate;
+        }
+      } catch (SQLException ignored) {
+        // intentionally ignore exception
+      }
+    }
+
+    throw new SQLException("Unable to find a driver that accepts url: " + realUrl);
+  }
+
+  private static String extractRealUrl(String url) {
+    return url.startsWith(INTERCEPTOR_MODE_URL_PREFIX)
+        ? url.replace(INTERCEPTOR_MODE_URL_PREFIX, "jdbc:")
+        : url;
+  }
+
+  @Nullable
+  @Override
+  public Connection connect(String url, Properties info) throws SQLException {
+    // if there is no url, we have problems
+    if (url == null) {
+      throw new SQLException("url is required");
+    }
+
+    if (!acceptsURL(url)) {
+      return null;
+    }
+
+    final String realUrl = extractRealUrl(url);
+
+    // find the real driver for the URL
+    final Driver wrappedDriver = findDriver(realUrl);
+
+    final Connection connection = wrappedDriver.connect(realUrl, info);
+
+    final DbInfo dbInfo = JdbcConnectionUrlParser.parse(realUrl, info);
+    JdbcMaps.connectionInfo.put(connection, dbInfo);
```

Comment:
There should be no need to use `JdbcMaps`in a library instrumentation: you can just add a `DbInfo` field to `OpenTelemetryConnection`.

Reply:
Done.